src/adapter/src/coord.rs

```diff
@@ -3332,6 +3331,126 @@ impl Coordinator {
         ]);
         Ok(serde_json::Value::Object(map))
     }
+
+    /// Checks that all clusters/collections are hydrated. If so, this will
+    /// trigger `self.clusters_hydrated_trigger`.
+    ///
+    /// This method is a no-op when the trigger has already been fired.
+    async fn maybe_check_hydration_status(&mut self) {
+        if self.clusters_hydrated_trigger.is_some() {
+            let enable_caught_up_check =
+                ENABLE_0DT_CAUGHT_UP_CHECK.get(self.catalog().system_config().dyncfgs());
+
+            if enable_caught_up_check {
+                // WIP: Not sure I like this one. Is there a smarter way of
+                // finding the collection.
+                let replica_frontier_collection_id = {
+                    let entry = self
+                        .catalog()
+                        .entries()
+                        .find(|entry| entry.name().item == "mz_cluster_replica_frontiers")
+                        .expect("missing mz_cluster_replica_frontiers");
+
+                    entry.id
+                };
```

Comment:
You could replace it with something like

```Rust
let id = self.catalog().resolve_builtin_table(&MZ_CLUSTER_REPLICA_FRONTIERS);
```

Reply:
🙌